packages/salesforcedx-vscode-core/src/settings/pushOrDeployOnSave.ts

```diff
@@ -176,8 +105,23 @@ function displayError(message: string) {
   channelService.showChannelOutput();
 }
 
-function ignorePath(uri: vscode.Uri) {
-  return isDotFile(uri) || isDirectory(uri);
+function ignorePath(uri: vscode.Uri, packageDirectories: string[]) {
+  return (
+    isDotFile(uri) ||
+    isDirectory(uri) ||
```

Comment:
Did you try using uri.scheme === file in `isDirectory` method ? It would be faster if we didn't have to use `fs`.